packages/smart-wallet/src/smartWallet.js

```diff
@@ -228,301 +235,288 @@ const behaviorGuards = {
-      const { brandPurses, paymentQueues: queues } = this.state;
-      const brand = await E(payment).getAllegedBrand();
-
-      // When there is a purse deposit into it
-      if (brandPurses.has(brand)) {
-        const purse = brandPurses.get(brand);
-        // @ts-expect-error deposit does take a FarRef<Payment>
-        return E(purse).deposit(payment);
-      }
-
-      // When there is no purse, queue the payment
-      if (queues.has(brand)) {
-        queues.get(brand).push(payment);
-      } else {
-        queues.init(brand, harden([payment]));
-      }
-      return AmountMath.makeEmpty(brand);
-    },
-  },
-  offers: {
-    /**
-     * @deprecated
-     * @returns {number} an error code, for backwards compatibility with clients expecting a number
-     */
-    getLastOfferId() {
-      return ERROR_LAST_OFFER_ID;
-    },
-    /**
-     * Take an offer description provided in capData, augment it with payments and call zoe.offer()
-     *
-     * @this {SmartWalletThis}
-     * @param {import('./offers.js').OfferSpec} offerSpec
-     * @returns {Promise<void>} when the offer has been sent to Zoe; payouts go into this wallet's purses
-     * @throws if any parts of the offer can be determined synchronously to be invalid
-     */
-    async executeOffer(offerSpec) {
-      const { facets } = this;
-      const {
-        address,
-        zoe,
-        brandPurses,
-        invitationBrand,
-        invitationPurse,
-        invitationIssuer,
-        offerToInvitationMakers,
-        offerToUsedInvitation,
-        updatePublishKit,
-      } = this.state;
-
-      const logger = {
-        info: (...args) => console.info('wallet', address, ...args),
-        error: (...args) => console.log('wallet', address, ...args),
-      };
-
-      const executor = makeOfferExecutor({
-        zoe,
-        depositFacet: facets.deposit,
-        invitationIssuer,
-        powers: {
-          invitationFromSpec: makeInvitationsHelper(
-            zoe,
-            invitationBrand,
-            invitationPurse,
-            offerToInvitationMakers.get,
-          ),
-          purseForBrand: brandPurses.get,
-          logger,
-        },
-        onStatusChange: offerStatus => {
-          logger.info('offerStatus', offerStatus);
-          updatePublishKit.publisher.publish({
-            updated: 'offerStatus',
-            status: offerStatus,
-          });
-        },
-        /** @type {(offerId: string, invitationAmount: Amount<'set'>, invitationMakers: object) => void} */
-        onNewContinuingOffer: (offerId, invitationAmount, invitationMakers) => {
-          offerToUsedInvitation.init(offerId, invitationAmount);
-          offerToInvitationMakers.init(offerId, invitationMakers);
-          facets.helper.publishCurrentState();
-        },
-      });
-      executor.executeOffer(offerSpec);
+const SmartWalletKit = defineVirtualFarClassKit(
+  'SmartWallet',
+  behaviorGuards,
+  initState,
+  {
+    helper: {
```

Comment:
somewhat regrettable to have this nested now, but it lets `this` be inferred automatically. There's not really any way to get that automated without making it a param in a function call with the state init function.